sql2.txt

```diff
@@ -17,29 +29,58 @@ may have ordered the same product in several different orders.  So the total qua
 SUM(Quantity) AS TotalQuantity and you must GROUP BY CustomerID, ProductID.  Paste your
 SQL statement below.
 
+SELECT Customers.CustomerName, Products.ProductName, SUM(OrderDetails.Quantity) AS TotalQuantity
+FROM Customers
+JOIN Orders on Customers.CustomerID = Orders.CustomerID                                       
+JOIN OrderDetails on Orders.OrderID = OrderDetails.OrderID
+JOIN Products on OrderDetails.ProductID = Products. ProductID
+GROUP BY Customers.CustomerID, OrderDetails. ProductID; 
+
+
 
 
 4. Extend the previous query to give the total price as well as the customer name, product name, 
 and total quantity.  Here you will have
 (SUM(Quantity) * Price) AS TotalPrice.  Paste your SQL statement below.
 
+SELECT Customers.CustomerName, Products.ProductName, SUM(OrderDetails.Quantity) AS TotalQuantity, (SUM(OrderDetails.Quantity)*Products.Price) AS TotalPrice
+FROM Customers
+JOIN Orders on Customers.CustomerID = Orders.CustomerID                                       
+JOIN OrderDetails on Orders.OrderID = OrderDetails.OrderID
+JOIN Products on OrderDetails.ProductID = Products. ProductID
+GROUP BY Customers.CustomerID, OrderDetails. ProductID; 
```

Comment:
Looks good, just same quick change needed as the previous answer:
1. You have an extra space in `Products. ProductID`.
2. The `ProductID` column needs a match the previous line and become `Products.ProductID`.